oxidation/crates/platform-async/src/platform/default.rs

```diff
@@ -0,0 +1,149 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use futures::FutureExt;
+use std::{
+    future::Future,
+    pin::Pin,
+    sync::{Arc, Mutex},
+    task::{Context, Poll, Waker},
+};
+
+/// Spawns a new asynchronous task, returning a [Task] for it.
+pub fn spawn<F>(future: F) -> Task<F::Output>
+where
+    F: Future + Send + 'static,
+    F::Output: Send + 'static,
+{
+    let shared_state = SharedState::default();
+    let shared_state = Arc::new(Mutex::new(shared_state));
+
+    let task = Task::new(shared_state.clone());
+    let runnable = Runnable::new(future, shared_state);
+
+    let _handle = tokio::spawn(runnable);
```

Comment:
would be useful in this kind of code to explain why dropping is okay

(btw, is there a rustonic way of showing drop without using the variable is not an error in the code ? using `drop(handle)` explicitly ?)

Reply:
clippy may don't like `drop(object)` https://rust-lang.github.io/rust-clippy/master/#drop_copy